chips/nrf51/src/pinmux.rs

```diff
@@ -0,0 +1,16 @@
+//! An abstraction over the NRF51 pin multiplexer
+//!
+//! Controller drivers should use the `Pinmux` type (instead of a `u32`) for
+//! fields that determine which pins are used by the hardware. The board
+//! configuration should create `Pinmux`s and pass them into controller drivers
+//! during initialization.
+
+/// An opaque wrapper around a configurable pin.
+#[derive(Copy, Clone)]
+pub struct Pinmux(u32);
+
```

Comment:
Is there any way in Rust for Pinmux to keep a static list of pin numbers that are already in use? Then if `Pinmux::new` is called with a repeated number it can `panic!` (or throw some type of error)